File: src/test/java/com/runninglane/dto/buddy/test/java/cases/contract/NonCompliantBaseClasses.java
package com.runninglane.dto.buddy.test.java.cases.contract;

public interface NonCompliantBaseClasses {
    interface InterfaceWithDefaultGetter {
        default String getName() {
            return "John Doe";
        }
    }

    interface InterfaceWithDefaultGetterAndAbstractSetter {
        default String getName() {
            return "John Doe";
        }

        void setName(String name);
    }

    interface InterfaceWithAbstractGetterAndDefaultSetter {
        String getName();

        default void setName(String name) {
            throw new UnsupportedOperationException("Not yet implemented");
        }
    }

    abstract class AbstractClassWithConcreteGetterAndAbstractSetter {
        public String getName() {
            return "John";
        }

        public abstract void setName(String name);
    }

    abstract class AbstractClassWithAbstractGetterAndConcreteSetter {
        public abstract String getName();

        public void setName(String name) {
            throw new UnsupportedOperationException("Not yet implemented");
        }
    }

    abstract class AbstractClassWithConcreteGetter {
        public String getName() {
            return "John Doe";
        }
    }

    abstract class AbstractClassWithConcreteSetter {
        public void setName(String name) {
            throw new UnsupportedOperationException("Not yet implemented");
        }
    }

}

File: src/test/kotlin/com/runninglane/dto/buddy/test/cases/contract/NonCompliantBaseClasses.kt
package com.runninglane.dto.buddy.test.cases.contract

interface InterfaceWithDefaultGetter {
    fun getName(): String = "John Doe"
}

interface InterfaceWithDefaultGetterAndAbstractSetter {
    fun getName(): String = "John Doe"
    fun setName(name: String)
}

interface InterfaceWithAbstractGetterAndDefaultSetter {
    fun getName(): String
    fun setName(name: String) { TODO("Not yet implemented") }
}

interface InterfaceWithAbstractPropertyAndDefaultGetter {
    val name: String
        get() = "John Doe"
}

abstract class AbstractClassWithConcreteGetterAndAbstractSetter {
    fun getName(): String = "John"
    abstract fun setName(name: String)
}

abstract class AbstractClassWithAbstractGetterAndConcreteSetter {
    abstract fun getName(): String
    fun setName(name: String) { TODO("Not yet implemented") }
}

abstract class AbstractClassWithConcreteGetter {
    fun getName(): String = "John Doe"
}

abstract class AbstractClassWithConcreteSetter {
    fun setName(name: String) { TODO("Not yet implemented") }
}

abstract class AbstractClassWithConcreteImmutableProperty {
    val name: String = "John Doe"
}
